### PR TITLE
[build] use createDatabase2

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

**Comment by cseed** [comment]
OK, looks like this will pass, ready for review.  We're going to have to rebuild all the databases when this goes in.

**Comment by cseed** [comment]
On, one thing, I removed delete_batch_database.  I don't quite know how to write it with the current setup.  I think we should add a new kind of step which is dev-only and only run if you ask for it explicitly (not just as a dependency).

**Comment by jigold** [comment]
I'm not sure what the best/easiest thing to do is. I think we can accomplish the same thing by copying and pasting the delete tables step (runImage) into a dev branch while testing. 

The other easiest thing I can think of is to add a `run_if_requested=True` option to each build step config and modify ci to skip over steps that aren't specifically requested in dev deploy. I don't think a new step is a good idea because what if I want an optional runImage step or an optional Deploy step, etc.

File: build.yaml [hunk 1]
```diff
@@ -72,6 +72,29 @@ steps:
...
+     cd repo
+     {{ code.checkout_script }}
+   outputs:
+     - from: /io/repo/auth/sql
+       to: /repo/auth/
+     - from: /io/repo/ci/sql
+       to: /repo/ci/
+     - from: /io/repo/batch/sql
+       to: /repo/batch/
+     - from: /io/repo/batch/sql
```
Comment (jigold) [inline build.yaml]: batch is repeated twice here

Comment (cseed) [reply build.yaml]: Thanks, fixed.

**Comment by cseed** [comment]
Sorry, we're in agreement here.  I was thinking a flag, too.

File: batch/sql/insert_globals.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+import os
 import string
 import secrets
 import asyncio
 from gear import Database
 
 
 async def main():
-    scope = sys.argv[1]
+    scope = os.environ['HAIL_SCOPE']
```
[the remark this answers is not in the document]
Comment (cseed) [reply batch/sql/insert_globals.py]: Yes, it is set in ci/create_database.py before calling the migration scripts.

File: build.yaml [hunk 3]
```diff
@@ -72,6 +72,27 @@ steps:
...
+     mkdir repo
+     cd repo
+     {{ code.checkout_script }}
+   outputs:
+     - from: /io/repo/auth/sql
+       to: /repo/auth/
+     - from: /io/repo/ci/sql
+       to: /repo/ci/
+     - from: /io/repo/batch/sql
+       to: /repo/batch/
```
[the remark this answers is not in the document]
Comment (cseed) [reply build.yaml]: It did not.  The existing code allows us to recursively copy directories.  His change was also to support archives to speed up copying directories with many files (e.g. the whole repo).

File: build.yaml [hunk 4]
```diff
@@ -124,53 +145,20 @@ steps:
...
-   dependsOn:
-    - default_ns
- - kind: buildImage
-   name: create_auth_tables_image
-   dockerFile: auth/Dockerfile.create-tables
-   contextPath: auth
-   dependsOn:
-    - base_image
- - kind: runImage
-   name: cleanup_auth_tables
```
[the remark this answers is not in the document]
Comment (cseed) [reply build.yaml]: I added it back.

**Pull request merged**